MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -193,345 +207,475 @@ void EnggDiffFittingPresenter::fittingFinished() {
-    if (strFPath.find("ENGINX_") != std::string::npos) {
-      boost::split(splitBaseName, strFPath, boost::is_any_of("_."));
-    }
+  // returns empty if no directory is found
+  const std::string parsedUserInput = pocoUserPathInput.toString();
 
-    // runNo when single focused file selected
-    std::vector<std::string> runNoVec;
+  // split directory if 'ENGINX_' found by '_.'
+  std::vector<std::string> splitBaseName;
+  if (parsedUserInput.find("ENGINX_") != std::string::npos) {
+    boost::split(splitBaseName, parsedUserInput, boost::is_any_of("_."));
+  }
 
+  try {
     // if input file is a directory and successfully splitBaseName
     // or when default bank is set or changed, the text-field is updated with
     // selected bank directory which would trigger this function again
-    if (selectedfPath.isFile() && !splitBaseName.empty()) {
-
-#ifdef __unix__
-      bankDir = selectedfPath.parent();
-#else
-      bankDir = (bankDir).expand(selectedfPath.parent().toString());
-#endif
-      // if vector is not empty and correct focus format file is selected
-      if (!splitBaseName.empty() && splitBaseName.size() > 3) {
+    if (pocoUserPathInput.isFile() && !splitBaseName.empty()) {
+      foundFullFilePaths =
+          processFullPathInput(pocoUserPathInput, splitBaseName);
+      // if given a multi-run
+    } else if (userPathInput.find("-") != std::string::npos) {
+      foundFullFilePaths = processMultiRun(userPathInput);
+      // try to process using single run
+    } else {
+      foundFullFilePaths = processSingleRun(userPathInput, splitBaseName);
+    }
+  } catch (std::invalid_argument &ia) {
+    // If something went wrong stop and print error only
+    g_log.error("Failed to process user input. Error was: ");
+    g_log.error(ia.what());
+    return;
+  }
 
-        std::string bankFileDir = bankDir.toString();
+  // if single or multi run-number
+  // set the text-field to directory here to the first in
+  // the vector if its not empty
+  if (foundFullFilePaths.empty()) {
+    m_view->userWarning(
+        "Error finding file(s)",
+        "Unable to find one or more files with a name matching the run"
+        " number or range input. Please check the files are present in"
+        " the focused output directory and the input is correct.");
+  } else if (!pocoUserPathInput.isFile()) {
+    // foundFiles is not empty and this is a directory
+    const std::string firstDir = foundFullFilePaths[0];
+    m_view->setFittingRunNo(firstDir);
+  }
+}
 
-        // browse the file
-        browsedFile(strFocusedFile, runnoDirVector, splitBaseName, runNoVec,
-                    bankFileDir);
-      }
-      // if given a multi-run OR single number
-    } else if (strFocusedFile.size() > 4) {
+/**
+  * Verifies and uses the user input path (i.e. a browsed file)
+  * to update drop down for available banks and various widgets on the GUI
+  *
+  * @param pocoFilePath The user entered file path as a Poco Path
```

Comment:
Are you using the name `pocoFilePath` to distinguish it from something else? It looks to me as if the `poco` part is just the type and could be omitted from the name.